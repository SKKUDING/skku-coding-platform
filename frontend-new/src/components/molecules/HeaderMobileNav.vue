<script setup lang="ts">
defineProps({
  active: {
    type: Boolean,
    default: false
  }
})

const items = [
  { label: 'Notices', link: '/notice' },
  { label: 'Contests', link: '/contest' },
  { label: 'Problems', link: '/problem' },
  { label: 'Lectures', link: '/lecture' }
]
</script>

<template>
  <div
    class="flex w-full flex-col items-center justify-center gap-2 overflow-hidden transition-all ease-out sm:flex-row sm:gap-4"
    :class="active ? 'h-48 border-b sm:h-12' : 'h-0'"
  >
    <router-link
      v-for="item in items"
      :key="item.label"
      :to="item.link"
      class="text-md font-semibold text-stone-600 transition hover:opacity-75 hover:ease-in-out active:opacity-50"
      active-class="text-lime-500"
    >
      {{ item.label }}
    </router-link>
    <div class="mt-2 flex gap-4 sm:hidden">
      <AtomsButton size="small" @click="$router.push('login')">
        Sign In
      </AtomsButton>
      <AtomsButton size="small" outline @click="$router.push('register')">
        Sign Up
      </AtomsButton>
    </div>
  </div>
</template>
