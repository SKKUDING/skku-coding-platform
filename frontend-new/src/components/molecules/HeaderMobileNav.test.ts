import { describe, test, expect } from 'vitest'
import { render } from '@testing-library/vue'
import HeaderMobileNav from './HeaderMobileNav.vue'

describe('Molecule - HeaderMobileNav.vue', () => {
  test('should render', () => {
    const { html } = render(HeaderMobileNav)
    expect(html()).toMatchSnapshot()
  })

  test('should show if active', () => {
    const { html } = render(HeaderMobileNav, {
      props: {
        active: true
      }
    })
    expect(html()).toMatchSnapshot()
  })
})
