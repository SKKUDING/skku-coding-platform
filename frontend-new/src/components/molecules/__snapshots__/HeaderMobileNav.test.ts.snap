// Vitest Snapshot v1

exports[`Molecule - HeaderMobileNav.vue > should render 1`] = `
"<div class=\\"flex flex-col sm:flex-row gap-2 sm:gap-4 items-center justify-center w-full overflow-hidden transition-all ease-out max-h-0\\">
  <router-link class=\\"text-lg font-semibold text-gray-600 hover:text-gray-400 active:text-gray-300 transition hover:ease-in-out\\" active-class=\\"
      text-lime-500
      hover:text-lime-400
      active:text-lime-300
    \\" to=\\"/notice\\">
    <!-- TODO: text color SKKU - #8DC63F --> Notices
  </router-link>
  <router-link class=\\"text-lg font-semibold text-gray-600 hover:text-gray-400 active:text-gray-300 transition hover:ease-in-out\\" active-class=\\"
      text-lime-500
      hover:text-lime-400
      active:text-lime-300
    \\" to=\\"/contest\\">
    <!-- TODO: text color SKKU - #8DC63F --> Contests
  </router-link>
  <router-link class=\\"text-lg font-semibold text-gray-600 hover:text-gray-400 active:text-gray-300 transition hover:ease-in-out\\" active-class=\\"
      text-lime-500
      hover:text-lime-400
      active:text-lime-300
    \\" to=\\"/problem\\">
    <!-- TODO: text color SKKU - #8DC63F --> Problems
  </router-link>
  <router-link class=\\"text-lg font-semibold text-gray-600 hover:text-gray-400 active:text-gray-300 transition hover:ease-in-out\\" active-class=\\"
      text-lime-500
      hover:text-lime-400
      active:text-lime-300
    \\" to=\\"/lecture\\">
    <!-- TODO: text color SKKU - #8DC63F --> Lectures
  </router-link>
</div>"
`;

exports[`Molecule - HeaderMobileNav.vue > should show if active 1`] = `
"<div class=\\"flex flex-col sm:flex-row gap-2 sm:gap-4 items-center justify-center w-full overflow-hidden transition-all ease-out max-h-40 py-2 border-b\\">
  <router-link class=\\"text-lg font-semibold text-gray-600 hover:text-gray-400 active:text-gray-300 transition hover:ease-in-out\\" active-class=\\"
      text-lime-500
      hover:text-lime-400
      active:text-lime-300
    \\" to=\\"/notice\\">
    <!-- TODO: text color SKKU - #8DC63F --> Notices
  </router-link>
  <router-link class=\\"text-lg font-semibold text-gray-600 hover:text-gray-400 active:text-gray-300 transition hover:ease-in-out\\" active-class=\\"
      text-lime-500
      hover:text-lime-400
      active:text-lime-300
    \\" to=\\"/contest\\">
    <!-- TODO: text color SKKU - #8DC63F --> Contests
  </router-link>
  <router-link class=\\"text-lg font-semibold text-gray-600 hover:text-gray-400 active:text-gray-300 transition hover:ease-in-out\\" active-class=\\"
      text-lime-500
      hover:text-lime-400
      active:text-lime-300
    \\" to=\\"/problem\\">
    <!-- TODO: text color SKKU - #8DC63F --> Problems
  </router-link>
  <router-link class=\\"text-lg font-semibold text-gray-600 hover:text-gray-400 active:text-gray-300 transition hover:ease-in-out\\" active-class=\\"
      text-lime-500
      hover:text-lime-400
      active:text-lime-300
    \\" to=\\"/lecture\\">
    <!-- TODO: text color SKKU - #8DC63F --> Lectures
  </router-link>
</div>"
`;
